sympy/combinatorics/fp_groups.py

```diff
@@ -277,6 +282,50 @@ def __str__(self):
     __repr__ = __str__
 
 
+class FpSubgroup(DefaultPrinting):
+    '''
+    The class implementing a subgroup of an FpGroup (only finite index subgroups are
+    supported at this point). This is to be used if one wishes to check if an element
+    of the original group belongs to the subgroup
+
+    '''
+    def __init__(self, G, gens):
+        super(FpSubgroup,self).__init__()
+        self.parent = G
+        self.generators = list(set([g for g in gens if g != G.identity]))
+        self.C = None
+
+    def __contains__(self, g):
+        if not g in self.parent:
+            raise ValueError
+        gens = g.contains_generators()
+        if any([g not in self.generators for g in gens]):
+            return False
+        if self.C is None:
+            C = self.parent.coset_enumeration(self.generators)
+            C.compress()
+            self.C = C
+        i = 0
+        C = self.C
+        for j in range(len(g)):
+            i = C.table[i][C.A_dict[g[j]]]
+        return i == 0
+
+    def order(self):
+        # This is valid because `len(self.C.table)` (the index of the subgroup)
+        # will always be finite - otherwise coset enumeration doesn't terminate
+        if not self.generators:
+            return 1
+        if self.C is None:
+            C = self.parent.coset_enumeration(self.generators)
+            C.compress()
+            self.C = C
+        return self.parent.order()/len(self.C.table)
+
+    def to_FpGroup(self):
+        return self.parent.subgroup(C=self.C)
+            
```

Comment:
Apparently there is trailing whitespace. (It is useful to configure the editor to show those. It is easy to forget `git diff --check` or `bin/test quality`.)